ios/FluentUI/Vnext/Button/MSFButton.swift

```diff
@@ -11,6 +11,15 @@ import UIKit
 @objc open class MSFButton: ControlHostingView,
                             UIGestureRecognizerDelegate {
 
+    open override func didUpdateFocus(in context: UIFocusUpdateContext, with coordinator: UIFocusAnimationCoordinator) {
+        let style = state.style
+        guard style == .primary || style == .accentFloating, let stateImpl = state as? MSFButtonStateImpl else {
```

Comment:
Instead of doing this type cast, you can create an internal constant of type MSFButtonStateImpl to store that reference directly:

`let stateImpl: MSFButtonStateImpl`

```
    @objc public init(style: MSFButtonStyle = .secondary,
                      size: MSFButtonSize = .large,
                      action: ((_ sender: MSFButton) -> Void)?) {
        self.action = action
        let buttonView = FluentButton(style: style,
                                      size: size,
                                      action: {})
        stateImpl = buttonView.state
...
    }
```

And make the state property a computed one that return this reference as the MSFButtonState protocol:

```
    @objc public var state: MSFButtonState {
        return stateImpl
    }
```